sdk/storage/src/blobs/block_blob_client.cpp

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "blobs/block_blob_client.hpp"
+
+#include "common/storage_common.hpp"
+
+namespace Azure { namespace Storage { namespace Blobs {
+
+  BlockBlobClient BlockBlobClient::FromConnectionString(
+      const std::string& connectionString,
+      const std::string& containerName,
+      const std::string& blobName,
+      const BlockBlobClientOptions& options)
+  {
+    BlockBlobClient newClient(
+        BlobClient::FromConnectionString(connectionString, containerName, blobName, options));
+    return newClient;
+  }
+
+  BlockBlobClient::BlockBlobClient(
+      const std::string& blobUri,
+      std::shared_ptr<SharedKeyCredential> credential,
+      const BlockBlobClientOptions& options)
+      : BlobClient(blobUri, std::move(credential), options)
+  {
+  }
+
+  BlockBlobClient::BlockBlobClient(
+      const std::string& blobUri,
+      std::shared_ptr<TokenCredential> credential,
+      const BlockBlobClientOptions& options)
+      : BlobClient(blobUri, std::move(credential), options)
+  {
+  }
+
+  BlockBlobClient::BlockBlobClient(
+      const std::string& blobUri,
+      const BlockBlobClientOptions& options)
+      : BlobClient(blobUri, options)
+  {
+  }
+
+  BlockBlobClient::BlockBlobClient(BlobClient blobClient) : BlobClient(std::move(blobClient))
+  {
+  }
+
+  BlockBlobClient BlockBlobClient::WithSnapshot(const std::string& snapshot)
+  {
+    BlockBlobClient newClient(*this);
+    if (snapshot.empty())
+    {
+      newClient.m_blobUrl.RemoveQuery("snapshot");
+    }
+    else
+    {
+      newClient.m_blobUrl.AppendQuery("snapshot", snapshot);
+    }
+    return newClient;
+  }
+
+  BlobContentInfo BlockBlobClient::Upload(
+      // TODO: We don't have BodyStream for now.
+      std::vector<uint8_t> content,
+      const UploadBlobOptions& options)
+  {
+    BlobRestClient::BlockBlob::UploadOptions protocolLayerOptions;
+    protocolLayerOptions.BodyBuffer = &content;
```

Comment:
>content [](start = 39, length = 7)

After the PR to change the request to hold the buffer, I think this needs to be changed as well to avoid the copy.